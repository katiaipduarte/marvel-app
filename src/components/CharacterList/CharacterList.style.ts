import styled from 'styled-components';

export const SearchOptions = styled.ul`
  list-style: none;
  height: auto;
  background: white;
  border-style: none;
  width: 75%;
  border-bottom-left-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  margin: 0 auto;
  padding: 1rem 0;

  li {
    height: 1rem;
    width: 100%;
    vertical-align: middle;
    margin: 0.5rem 0;
    cursor: pointer;

    p {
      width: inherit;
      padding: 0.25rem 0;
      font-size: 1rem;
      font-weight: 500;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      cursor: pointer;
    }

    &:hover,
    &:focus,
    &:focus-visible {
      p {
        color: ${({ theme }) => theme.primaryColor};
      }
    }
  }

  @media (min-width: 768px) {
    max-width: 32rem;
  }
`;
