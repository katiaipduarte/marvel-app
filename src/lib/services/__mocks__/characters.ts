import { Character } from '../../../interfaces/character';

/* eslint-disable max-len */
export const mockedCharacters: Character[] = [
  {
    id: 1009610,
    name: 'Spider-Man',
    description:
      'Bitten by a radioactive spider, high school student Peter Parker gained the speed, strength and powers of a spider. Adopting the name Spider-Man, Peter hoped to start a career using his new abilities. Taught that with great power comes great responsibility, Spidey has vowed to use his powers to help people.',
    modified: '2020-07-21T10:30:10-0400',
    thumbnail: { path: 'http://i.annihil.us/u/prod/marvel/i/mg/3/50/526548a343e4b', extension: 'jpg' },
    resourceURI: 'http://gateway.marvel.com/v1/public/characters/1009610',
    comics: {
      available: 4021,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009610/comics',
      items: [
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/62304',
          name: 'Spider-Man: 101 Ways to End the Clone Saga (1997) #1',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/78503', name: '2099 Alpha (2019) #1' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/60151',
          name: 'A YEAR OF MARVELS TPB (Trade Paperback)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/57926',
          name: 'A Year of Marvels: April Infinite Comic (2016) #1',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/57924',
          name: 'A Year of Marvels: February Infinite Comic (2016) #1',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/43501', name: 'A+X (2012) #4' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/76012', name: 'Absolute Carnage (2019) #1' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/76014', name: 'Absolute Carnage (2019) #2' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/78960', name: 'Absolute Carnage (2019) #5' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/77071',
          name: 'Absolute Carnage: Symbiote Spider-Man (2019) #1',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/320',
          name: 'Actor Presents Spider-Man and the Incredible Hulk (2003) #1',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76863',
          name: 'Adventures of Spider-Man (1996) #1',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76864',
          name: 'Adventures of Spider-Man (1996) #2',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76865',
          name: 'Adventures of Spider-Man (1996) #3',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76866',
          name: 'Adventures of Spider-Man (1996) #4',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76867',
          name: 'Adventures of Spider-Man (1996) #5',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76868',
          name: 'Adventures of Spider-Man (1996) #6',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76869',
          name: 'Adventures of Spider-Man (1996) #7',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76870',
          name: 'Adventures of Spider-Man (1996) #8',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/76871',
          name: 'Adventures of Spider-Man (1996) #9',
        },
      ],
      returned: 20,
    },
    series: {
      available: 1030,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009610/series',
      items: [
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/26024',
          name: ' Superior Spider-Man Vol. 2: Otto-matic (2019)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/27980', name: '2099 Alpha (2019)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/22102', name: 'A YEAR OF MARVELS TPB (2017)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/27342',
          name: 'A Year of Marvels: April Infinite Comic (2016)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/25677',
          name: 'A Year of Marvels: February Infinite Comic (2016)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/16450', name: 'A+X (2012 - 2014)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/27272',
          name: 'Absolute Carnage (2019 - Present)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/27637',
          name: 'Absolute Carnage: Symbiote Spider-Man (2019)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/458',
          name: 'Actor Presents Spider-Man and the Incredible Hulk (2003)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/27587',
          name: 'Adventures of Spider-Man (1996 - 1997)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/27025',
          name: 'Adventures Of Spider-Man: Sinister Intentions (2019)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/27671',
          name: 'Adventures Of Spider-man: Spectacular Foes (2019)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/9790', name: 'Age of Heroes (2010)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/10235', name: 'AGE OF HEROES TPB (2011)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/17318', name: 'Age of Ultron (2013)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/24134',
          name: 'Agents of Atlas: The Complete Collection Vol. 1 (2018)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/672', name: 'Alias (2001 - 2003)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/13383', name: 'Alias Omnibus (2006)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/2116', name: 'Alpha Flight (1983 - 1994)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/17650', name: 'Alpha: Big Time (2013)' },
      ],
      returned: 20,
    },
    stories: {
      available: 6008,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009610/stories',
      items: [
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/483',
          name: 'Interior #483',
          type: 'interiorStory',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/486', name: 'Cover #486', type: 'cover' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/487',
          name: 'Interior #487',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/498',
          name: 'SENSATIONAL SPIDER-MAN (2006) #23',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/499',
          name: 'Interior #499',
          type: 'interiorStory',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/598', name: 'Cover #598', type: 'cover' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/599',
          name: 'Interior #599',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/805',
          name: 'Interior #805',
          type: 'interiorStory',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/824', name: 'Cover #824', type: 'cover' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/838',
          name: '3 of 3 - Family Business',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/842',
          name: '1 of 1 -  Secret of the Spider Shop',
          type: 'cover',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/866', name: 'Cover #866', type: 'cover' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/912',
          name: 'Fantastic Four (1998) #512',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/913',
          name: 'Interior #913',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/916',
          name: 'Fantastic Four (1998) #513',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/917',
          name: 'Interior #917',
          type: 'interiorStory',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/977', name: 'Cover #977', type: 'cover' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/979', name: 'Cover #979', type: 'cover' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/1018',
          name: 'Amazing Spider-Man (1999) #500',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/1019',
          name: 'Interior #1019',
          type: 'interiorStory',
        },
      ],
      returned: 20,
    },
    events: {
      available: 38,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009610/events',
      items: [
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/116', name: 'Acts of Vengeance!' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/314', name: 'Age of Ultron' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/233', name: 'Atlantis Attacks' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/234', name: 'Avengers Disassembled' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/310', name: 'Avengers VS X-Men' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/296', name: 'Chaos War' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/238', name: 'Civil War' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/318', name: 'Dark Reign' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/240', name: 'Days of Future Present' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/events/332',
          name: 'Dead No More: The Clone Conspiracy',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/245', name: 'Enemy of the State' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/246', name: 'Evolutionary War' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/302', name: 'Fear Itself' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/251', name: 'House of M' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/252', name: 'Inferno' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/315', name: 'Infinity' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/317', name: 'Inhumanity' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/255', name: 'Initiative' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/258', name: "Kraven's Last Hunt" },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/151', name: 'Maximum Carnage' },
      ],
      returned: 20,
    },
    urls: [
      {
        type: 'detail',
        url: 'http://marvel.com/characters/54/spider-man?utm_campaign=apiRef&utm_source=07e3e205bebd46de31d15ee9a76d85c2',
      },
      {
        type: 'wiki',
        url: 'http://marvel.com/universe/Spider-Man_(Peter_Parker)?utm_campaign=apiRef&utm_source=07e3e205bebd46de31d15ee9a76d85c2',
      },
      {
        type: 'comiclink',
        url: 'http://marvel.com/comics/characters/1009610/spider-man?utm_campaign=apiRef&utm_source=07e3e205bebd46de31d15ee9a76d85c2',
      },
    ],
  },
  {
    id: 1009189,
    name: 'Black Widow',
    description: '',
    modified: '2016-01-04T18:09:26-0500',
    thumbnail: { path: 'http://i.annihil.us/u/prod/marvel/i/mg/f/30/50fecad1f395b', extension: 'jpg' },
    resourceURI: 'http://gateway.marvel.com/v1/public/characters/1009189',
    comics: {
      available: 567,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009189/comics',
      items: [
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/43495', name: 'A+X (2012) #2' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/43509', name: 'A+X (2012) #10' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/22461',
          name: 'Adam: Legend of the Blue Marvel (2008) #1',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23245', name: 'Amazing Adventures (1970) #1' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23256', name: 'Amazing Adventures (1970) #2' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23267', name: 'Amazing Adventures (1970) #3' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23278', name: 'Amazing Adventures (1970) #4' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23279', name: 'Amazing Adventures (1970) #5' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23280', name: 'Amazing Adventures (1970) #6' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23281', name: 'Amazing Adventures (1970) #7' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/23282', name: 'Amazing Adventures (1970) #8' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/40109',
          name: 'Amazing Spider-Man (1999) #684',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/40116',
          name: 'Amazing Spider-Man (1999) #685',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/40115',
          name: 'Amazing Spider-Man (1999) #686',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/comics/40113',
          name: 'Amazing Spider-Man (1999) #687',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/17490', name: 'Avengers (1998) #1' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/17501', name: 'Avengers (1998) #2' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/17512', name: 'Avengers (1998) #3' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/17523', name: 'Avengers (1998) #4' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/comics/17509', name: 'Avengers (1998) #27' },
      ],
      returned: 20,
    },
    series: {
      available: 214,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009189/series',
      items: [
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/16450', name: 'A+X (2012 - 2014)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/6079',
          name: 'Adam: Legend of the Blue Marvel (2008)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/17318', name: 'Age of Ultron (2013)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/6666',
          name: 'Amazing Adventures (1970 - 1976)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/454',
          name: 'Amazing Spider-Man (1999 - 2013)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/354', name: 'Avengers (1998 - 2004)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/1991', name: 'Avengers (1963 - 1996)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/15373',
          name: 'Avengers Assemble (2012 - 2014)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/1340', name: 'Avengers Assemble (2004)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/1737',
          name: 'Avengers Assemble Vol. 3 (2006)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/1816',
          name: 'Avengers Assemble Vol. 4 (2007)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/2384',
          name: 'Avengers Classic (2007 - 2008)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/157',
          name: 'Avengers Legends Vol. 2: The Korvac Saga (2003)',
        },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/22229', name: 'Avengers VS (2016)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/15305', name: 'Avengers Vs. X-Men (2012)' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/series/18398', name: 'Avengers World (2014 - 2015)' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/4864',
          name: 'Avengers/Invaders (2008 - 2009)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/937',
          name: "Avengers: Earth's Mightiest Heroes II (2006 - 2007)",
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/17881',
          name: 'Avengers: Endless Wartime (2013)',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/series/25869',
          name: 'Avengers: Shards of Infinity (2018)',
        },
      ],
      returned: 20,
    },
    stories: {
      available: 650,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009189/stories',
      items: [
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/1060', name: 'Cover #1060', type: 'cover' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/stories/1066', name: 'Cover #1066', type: 'cover' },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/1068',
          name: '"THE WIDOW," PART 4 (OF 4) The blistering climax to the sexy 4-part Black Widow storyline (just in time for a certain blistering',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/1092',
          name: '2 of 5 - Murdock Papers',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/1094',
          name: '3 of 5 - Murdock Papers',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/1096',
          name: '4 of 6 - Murdock Papers',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/1098',
          name: '5 of 6 - Murdock Papers',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3455',
          name: '6 of 6 - Breakout!',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3535',
          name: 'Black Widow (2004) #3',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3539',
          name: 'Black Widow (2004) #2',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3540',
          name: 'Black Widow (2004) #4',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3541',
          name: 'Black Widow (2004) #4',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3543',
          name: 'Black Widow (2004) #5',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3545',
          name: 'Black Widow (2004) #6',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/3560',
          name: 'ULTIMATE NIGHTMARE (2004) #2',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/4113',
          name: '1 of 1 - Captain America & Black Widow',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/5185',
          name: "Avengers: Earth's Mightiest Heroes II (2006) #5",
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/5186',
          name: '5 of 8 - 8XLS',
          type: 'interiorStory',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/6044',
          name: 'Wolverine Origins (2006) #8',
          type: 'cover',
        },
        {
          resourceURI: 'http://gateway.marvel.com/v1/public/stories/7750',
          name: "Avengers: Earth's Mightiest Heroes II (2006) #8",
          type: 'cover',
        },
      ],
      returned: 20,
    },
    events: {
      available: 16,
      collectionURI: 'http://gateway.marvel.com/v1/public/characters/1009189/events',
      items: [
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/116', name: 'Acts of Vengeance!' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/314', name: 'Age of Ultron' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/310', name: 'Avengers VS X-Men' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/296', name: 'Chaos War' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/248', name: 'Fall of the Mutants' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/302', name: 'Fear Itself' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/315', name: 'Infinity' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/317', name: 'Inhumanity' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/255', name: 'Initiative' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/37', name: 'Maximum Security' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/154', name: 'Onslaught' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/319', name: 'Original Sin' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/336', name: 'Secret Empire' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/269', name: 'Secret Invasion' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/270', name: 'Secret Wars' },
        { resourceURI: 'http://gateway.marvel.com/v1/public/events/309', name: 'Shattered Heroes' },
      ],
      returned: 16,
    },
    urls: [
      {
        type: 'detail',
        url: 'http://marvel.com/comics/characters/1009189/black_widow?utm_campaign=apiRef&utm_source=07e3e205bebd46de31d15ee9a76d85c2',
      },
      {
        type: 'wiki',
        url: 'http://marvel.com/universe/Black_Widow_(Natasha_Romanova)?utm_campaign=apiRef&utm_source=07e3e205bebd46de31d15ee9a76d85c2',
      },
      {
        type: 'comiclink',
        url: 'http://marvel.com/comics/characters/1009189/black_widow?utm_campaign=apiRef&utm_source=07e3e205bebd46de31d15ee9a76d85c2',
      },
    ],
  },
];
