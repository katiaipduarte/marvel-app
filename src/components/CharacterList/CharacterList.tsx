import { Character } from '../../interfaces/character';
import { SearchOptions } from './CharacterList.style';

type Props = {
  characters: Character[];
  selectCharacter: (id: number, name: string) => void;
};

const CharacterList = (props: Props): JSX.Element => {
  const { characters, selectCharacter } = props;

  return (
    <SearchOptions role="list" tabIndex={-1}>
      {characters.length === 0 ? (
        <li data-testid="empty-list">No results found</li>
      ) : (
        <>
          {characters.map((character: Character) => (
            <li
              key={character.id}
              onClick={() => selectCharacter(character.id, character.name)}
              onKeyPress={() => selectCharacter(character.id, character.name)}
              tabIndex={0}
              role="button"
            >
              <p data-testid="character-name">{character.name}</p>
            </li>
          ))}
        </>
      )}
    </SearchOptions>
  );
};

export default CharacterList;
