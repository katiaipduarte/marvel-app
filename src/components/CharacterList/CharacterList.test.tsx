import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { mockedCharacters } from '../../lib/services/__mocks__/characters';
import CharacterList from './CharacterList';

describe('the <CharacterList /> component', () => {
  const selectCharacter = jest.fn();

  it('should render correctly a "No results found" message', () => {
    const { container } = render(<CharacterList selectCharacter={selectCharacter} characters={[]} />);
    expect(container.firstChild).toMatchSnapshot();

    const text = screen.getByTestId('empty-list');
    expect(text).toBeInTheDocument();
  });

  it('should render a list with characters', () => {
    const mockedData = mockedCharacters;

    const { container } = render(<CharacterList selectCharacter={selectCharacter} characters={mockedData} />);

    const characterNames = screen.queryAllByTestId('character-name');

    expect(characterNames.length).toBe(2);
    expect(characterNames[0].innerHTML).toEqual('Spider-Man');
    expect(characterNames[1].innerHTML).toEqual('Black Widow');
    expect(container.firstChild).toMatchSnapshot();
  });
});
