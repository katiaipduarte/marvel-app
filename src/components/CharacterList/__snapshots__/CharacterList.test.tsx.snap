// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <CharacterList /> component should render a list with characters 1`] = `
.c0 {
  list-style: none;
  height: auto;
  background: white;
  border-style: none;
  width: 75%;
  border-bottom-left-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  padding: 1rem 0;
  position: absolute;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  text-align: center;
  z-index: 1;
  top: 7.45rem;
}

.c0 li {
  height: 1rem;
  width: 100%;
  vertical-align: middle;
  margin: 0.5rem 0;
  cursor: pointer;
}

.c0 li p {
  width: inherit;
  padding: 0.25rem 0;
  font-size: 1rem;
  font-weight: 500;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
}

@media (min-width:768px) {
  .c0 {
    max-width: 32rem;
  }
}

<ul
  class="c0"
  role="list"
  tabindex="-1"
>
  <li
    role="button"
    tabindex="0"
  >
    <p
      data-testid="character-name"
    >
      Spider-Man
    </p>
  </li>
  <li
    role="button"
    tabindex="0"
  >
    <p
      data-testid="character-name"
    >
      Black Widow
    </p>
  </li>
</ul>
`;

exports[`the <CharacterList /> component should render correctly a "No results found" message 1`] = `
.c0 {
  list-style: none;
  height: auto;
  background: white;
  border-style: none;
  width: 75%;
  border-bottom-left-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  padding: 1rem 0;
  position: absolute;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  text-align: center;
  z-index: 1;
  top: 7.45rem;
}

.c0 li {
  height: 1rem;
  width: 100%;
  vertical-align: middle;
  margin: 0.5rem 0;
  cursor: pointer;
}

.c0 li p {
  width: inherit;
  padding: 0.25rem 0;
  font-size: 1rem;
  font-weight: 500;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
}

@media (min-width:768px) {
  .c0 {
    max-width: 32rem;
  }
}

<ul
  class="c0"
  role="list"
  tabindex="-1"
>
  <li
    data-testid="empty-list"
  >
    No results found
  </li>
</ul>
`;
